server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -476,11 +476,26 @@ public long ramBytesUsed() {
     }
 
     /**
-     * Returns how much RAM would be freed up by refreshing. This is the RAM usage of the current version map. It doesn't include tombstones
-     * since they don't get cleared on refresh, nor the old version map that is being reclaimed.
+     * Returns how much RAM is used by refresh. This is the RAM usage of the current and old version maps, and the RAM usage of the
+     * archive, if any.
      */
     long ramBytesUsedForRefresh() {
-        return maps.current.ramBytesUsed.get();
+        return maps.ramBytesUsed() + archive.getMemoryBytesUsed();
+    }
+
+    /**
+     * Returns how much RAM could be reclaimed from the version map.
+     * <p>
+     * In stateful, this is the RAM usage of the current version map, and could be reclaimed by refreshing. It doesn't include tombstones
+     * since they don't get cleared on refresh, nor the old version map that is being reclaimed.
+     * <p>
+     * In stateless, this is the RAM usage of current and old version map plus the RAM usage of the parts of the archive that require a
+     * new unpromotable refresh. To reclaim all three components we need to refresh AND flush.
+     */
```

Comment:
Also for the Stateless `reclaimableRamBytes`, I have not considered the tombstones, since it is not entirely cleaned up on a flush and also depends on some interval settings (`INDEX_GC_DELETES_SETTING`). We could consider it as part of `reclaimableRamBytes`, but I think it might be not easy to report how much of it will be part of `getRefreshingBytes`. So I have decided not to include it.